### ISSUE TITLE
Incomprehensible warning when using recursive generics in IntelliJ IDEA

### ISSUE THREAD

**Issue by Lignium** [typetools/checker-framework#4902]
When using recursive generics, for example, creating a self-reference, an incomprehensible warning is raised. When using simple non-recursive generics, there is no warning. Perhaps I do not understand something? Maybe this is an IDE bug?

`Experiment.java`
```java
package org.lignium.experiment;

public final class Experiment {
    private static abstract class Region<R extends Region<R>> {
        public abstract R self();
    }
    private static final class CuboidRegion extends Region<CuboidRegion> {
        @Override
        public CuboidRegion self() {
            return this;
        }
    }
    public static void main(String[] arguments) {
        // Warning at wildcard: Non-null type argument is expected
        Region<?> region = new CuboidRegion();
    }
}
```

`package-info.java`
```java
@DefaultQualifier(NonNull.class)
package org.lignium.experiment;

import org.checkerframework.checker.nullness.qual.NonNull;
import org.checkerframework.framework.qual.DefaultQualifier;
```

**Comment by smillst** [comment]
I can't reproduce any error for this example. 

```
javacheck -processor nullness Experiment.java package-info.java
```

**Comment by Lignium** [comment]
Is it an IntelliJ IDEA bug?

**Comment by smillst** [comment]
The warning you mention, `Non-null type argument is expected`, is produced by IntelliJ IDEA not the Nullness Checker. Since I can't reproduce the error, I can't say if it's an IntelliJ IDEA bug.

**Comment by mernst** [comment]
Since the error is produced by IntelliJ IDEA, I'm closing this issue.
